src/appshell/view/framelesswindow/framelesswindowmodel.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "framelesswindowmodel.h"
+
+#include <QWindow>
+
+#ifdef Q_OS_WIN
+#include <dwmapi.h>
+#include "windowsnativeeventfilter.h"
+#else
+#include "abstractnativeeventfilter.h"
+#endif
+
+using namespace mu::appshell;
+
+FremelessWindowModel::FremelessWindowModel(QObject* parent)
+    : QObject(parent)
+{
+#ifdef Q_OS_WIN
+    m_nativeEventFilter = new WindowsNativeEventFilter();
+#else
+    m_nativeEventFilter = new AbstractNativeEventFilter();
+#endif
+
+    qApp->installNativeEventFilter(m_nativeEventFilter);
+}
+
+FremelessWindowModel::~FremelessWindowModel()
+{
+    delete m_nativeEventFilter;
+}
+
+void FremelessWindowModel::init(QWindow* window)
+{
+#ifdef Q_OS_WIN
+    HWND hwnd = (HWND)window->winId();
+
+    SetWindowLongPtr(hwnd, GWL_STYLE,
+                     static_cast<LONG>(WS_POPUP | WS_CAPTION | WS_SYSMENU | WS_MINIMIZEBOX | WS_MAXIMIZEBOX | WS_THICKFRAME
+                                       | WS_CLIPCHILDREN));
+
+    const MARGINS shadow_on = { 1, 1, 1, 1 };
+    DwmExtendFrameIntoClientArea(hwnd, &shadow_on);
+
+    SetWindowPos(hwnd, Q_NULLPTR, 0, 0, 0, 0, SWP_FRAMECHANGED | SWP_NOMOVE | SWP_NOSIZE);
+    ShowWindow(hwnd, SW_SHOW);
+#endif
```

Comment:
I think, better AbstractNativeEventFilter rename to FramelessWindowController and put this code to it.
qApp->installNativeEventFilter(m_nativeEventFilter); put to it too, so will be 
```
void FremelessWindowModel::init(QWindow* window)
{
    m_controller->init(window);
}
```

void WinFremelessWindowController::init(QWindow* window)
{
      qApp->installNativeEventFilter(this); 
      HWND hwnd = (HWND)window->winId();
      ...
}

Reply:
thanks! fixed